js/functions.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const querystring = require('querystring');
+
+module.exports = (config = {}) => {
+
+  config = Object.assign({
+    add_version: true,
+  }, config);
+
+  return {
+    asset: (asset_path) => {
+      if (!config.path) {
+        return asset_path;
+      }
+      const absolute_path = path.join(config.path, asset_path);
+      if (fs.existsSync(absolute_path) && config.add_version) {
```

Comment:
Avoid the expensive operation if there are more simple conditions.
```suggestion
      if (config.add_version && fs.existsSync(absolute_path)) {
```